src/index/src/fulltext_index/create/tantivy.rs

```diff
@@ -59,10 +68,17 @@ impl TantivyFulltextIndexCreator {
 
         let memory_limit = Self::sanitize_memory_limit(memory_limit);
 
-        let writer = SingleSegmentIndexWriter::new(index, memory_limit).context(TantivySnafu)?;
```

Comment:
The comment above said that `the maximum capacity for the index is limited to 2<<31 rows (around 2 billion rows)`. Why do we get an error when there are 50M rows?

Reply:
An internal buffer has a size limit. `SingleSegmentIndexWriter` is a dragon API and does not pay attention to this size limit, resulting in an error after exceeding it.

Reply:
🐉

Reply:
🐉